rotkehlchen/api/rest.py

```diff
@@ -4416,3 +4417,41 @@ def search_for_names_everywhere(self, addresses: List[ChecksumEthAddress]) -> Re
             addresses=addresses,
         )
         return api_response(_wrap_in_ok_result(mappings))
+
+    def _detect_ethereum_tokens(
+            self,
+            addresses: List[ChecksumEthAddress],
+    ) -> Dict[str, Any]:
+        ethtokens = EthTokens(
+            database=self.rotkehlchen.data.db,
+            ethereum=self.rotkehlchen.chain_manager.ethereum,
+        )
+        try:
+            with self.rotkehlchen.data.db.user_write() as cursor:
+                ethtokens.detect_tokens(
+                    write_cursor=cursor,
+                    accounts=addresses,
+                )
+        except (RemoteError, BadFunctionCallOutput) as e:
+            return wrap_in_fail_result(message=str(e), status_code=HTTPStatus.CONFLICT)
+        return OK_RESULT
+
+    def detect_ethereum_tokens(
+            self,
+            async_query: bool,
+            addresses: Optional[List[ChecksumEthAddress]],
+    ) -> Response:
+        if addresses is None:
+            addresses = self.rotkehlchen.chain_manager.accounts.eth
+        if async_query is True:
+            return self._query_async(
+                command=self._detect_ethereum_tokens,
+                addresses=addresses,
+            )
+
+        response = self._detect_ethereum_tokens(
+            addresses=addresses,
+        )
```

Comment:
If code fits in one line put it there
```suggestion
        response = self._detect_ethereum_tokens(addresses=addresses)
```

Reply:
Done